lib/src/components/NxTag/NxTag.scss

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @import '../../scss-shared/nx-variables';
+ @import '../../scss-shared/nx-text-helpers';
+ @import '../../scss-shared/nx-container-helpers';
+
+.nx-tag {
+  @include semi-bold();
+  @include container-horizontal;
+  @include nx-truncate-ellipsis;
+  background-color: $nx-blue-800;
+  border: 1px solid $nx-blue-800;
+  border-radius: 16px;
+  box-sizing: border-box;
+  color: $nx-text-color-dark;
+  cursor: default;
+  display: inline-block;
+  font-size: $nx-font-size-s;
+  height: 32px;
+  margin: 0 $nx-spacing-xs $nx-spacing-xs 0;
+  padding: 5px $nx-spacing-md 6px $nx-spacing-md;
+  // Uppercase Min so SCSS ignores it and uses CSS min, not SCSS min
+  max-width: Min(100%, 320px);
+}
+
+.nx-tag--selectable {
+  align-items: center;
+  cursor: pointer;
+  display: inline-flex;
+  padding-right: $nx-spacing-xs;
+
+  &:focus {
+    border: 1px solid $nx-focus-border-color;
+    box-shadow: $nx-focus-box-shadow;
+    outline: 0;
+  }
+
+  &:hover {
+    border-color: $nx-grey-900;
+  }
+}
+
+.nx-tag--selected {
+  background-color: $nx-blue-800;
+  border-color: $nx-blue-800;
+  color: $nx-text-color-dark;
+
+  &:focus {
+    border: 1px solid $nx-focus-border-color;
+  }
+}
+
+.nx-tag--unselected {
+  border-color: $nx-grey-300;
+}
+
+.nx-tag__text {
+  @include nx-truncate-ellipsis;
+}
+
+.nx-tag__action {
+  margin-left: $nx-spacing-md;
+}
+
+// Available tag colours
+.nx-tag--default {
```

Comment:
I feel like there must be some redundancy in the way you've got the colors specified.  You've got the colored tags specified using the mixin below, and the default tag specified here.  But you are also specifying values for the background, border, and text color in multiple places above this.  Do those ones actually cover any cases or do they always end up overridden?

Reply:
> I feel like there must be some redundancy in the way you've got the colors specified. You've got the colored tags specified using the mixin below, and the default tag specified here. But you are also specifying values for the background, border, and text color in multiple places above this. Do those ones actually cover any cases or do they always end up overridden?

Part of the issue is the funky default colour specified by @dturner32. If we get that cleaned up I think I'll probably be able to clean this up a bit.

Reply:
> Part of the issue is the funky default colour specified by @dturner32. If we get that cleaned up I think I'll probably be able to clean this up a bit.

I don't think that's really the issue.  We've got:

1. Background, border, and text color all specified directly on `.nx-tag`
2. Background, border, and text color all specified directly on `.nx-tag--selected`
3. Border color specified directly on `.nx-tag--unselected`
4. Text color specified on `.nx-tag--default`.  This would seem to be redundant with one of the above.
5. Background and text color specified on `.nx-tag--default.nx-tag--unselected`.  This would seem to be redundant with 1 and/or 3 above.
6. Background and border color specified on each `.nx-tag--#{color}`
7. Background specified on each `.nx-tag--#{color}.nx-tag--unselected`.

6 and 7 are fine, but 1, 2, and 3 all seem to be mostly redundant with 4 and 5.

IMO a clearer way to organize it would be to remove default-tag-specific colors (such as the blue background) from the `.nx-tag` and `.nx-tag--unselected` styles.  Only have the near-universal colors on `nx-tag` - effectively just the dark text color.  Then have each `.nx-tag--#{color}`, and `.nx-tag--default` and the `unselected` modifiers of them, all specify their background and border colors, and also have `.nx-tag--default.nx-tag--selected` specify its oddball text color.

Reply:
> > Part of the issue is the funky default colour specified by @dturner32. If we get that cleaned up I think I'll probably be able to clean this up a bit.
> 
> I don't think that's really the issue. We've got:
> 
> 1. Background, border, and text color all specified directly on `.nx-tag`
> 2. Background, border, and text color all specified directly on `.nx-tag--selected`
> 3. Border color specified directly on `.nx-tag--unselected`
> 4. Text color specified on `.nx-tag--default`.  This would seem to be redundant with one of the above.
> 5. Background and text color specified on `.nx-tag--default.nx-tag--unselected`.  This would seem to be redundant with 1 and/or 3 above.
> 6. Background and border color specified on each `.nx-tag--#{color}`
> 7. Background specified on each `.nx-tag--#{color}.nx-tag--unselected`.
> 
> 6 and 7 are fine, but 1, 2, and 3 all seem to be mostly redundant with 4 and 5.
> 
> IMO a clearer way to organize it would be to remove default-tag-specific colors (such as the blue background) from the `.nx-tag` and `.nx-tag--unselected` styles. Only have the near-universal colors on `nx-tag` - effectively just the dark text color. Then have each `.nx-tag--#{color}`, and `.nx-tag--default` and the `unselected` modifiers of them, all specify their background and border colors, and also have `.nx-tag--default.nx-tag--selected` specify its oddball text color.

With Derrek's new colours this has all be cleaned up in: https://github.com/sonatype/sonatype-react-shared-components/pull/290/commits/536730212837c4311e2c812c0d5c43e67da8a591